src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -194,7 +195,7 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
             nuint offset = 0; // Use nuint for arithmetic to avoid unnecessary 64->32->64 truncations
             nuint lengthToExamine = (nuint)(uint)length;
 
-            if (Avx2.IsSupported || Sse2.IsSupported)
+            if (Avx2.IsSupported || Sse2.IsSupported || AdvSimd.IsSupported)
```

Comment:
I know it's not part of your PR, but this check could just be: 
```suggestion
            if (Sse2.IsSupported || AdvSimd.IsSupported)
```

`Avx2.IsSupported` can only be true if `Sse2.IsSupported` is also true

Reply:
Sure, fixed here and at other palces.